src/coreclr/vm/class.cpp

```diff
@@ -2860,6 +2860,34 @@ void SparseVTableMap::RecordGap(WORD StartMTSlot, WORD NumSkipSlots)
     m_MapEntries++;
 }
 
+//*******************************************************************************
+// While building mapping list, record an excluded MT slot.
+void SparseVTableMap::RecordExcludedMethod(WORD MTSlot)
```

Comment:
I don't see how this API is any different than the existing by calling `RecordGap(x, -1)`. I appreciate we've got less/different pre-conditions, but the general logic is identical. The minor difference is that passing `-1` is an invalid precondition, but the intent for both is we are changing slots. Can we collapse the map update logic between the two? Even if this API just calls `RecordGap(MTSlot, -1)`? It would require updating an assert in `RecordGap`, but keeps the map update logic in a single place.